src/build.cc

```diff
@@ -1067,11 +1068,11 @@ bool Builder::ExtractDeps(CommandRunner::Result* result,
       // complexity in IncludesNormalize::Relativize.
       deps_nodes->push_back(state_->GetNode(*i, ~0u));
     }
-  } else
-  if (deps_type == "gcc") {
+  } else if (deps_type == "gcc" || deps_type == "msvc_source_dependencies") {
     string depfile = result->edge->GetUnescapedDepfile();
     if (depfile.empty()) {
-      *err = string("edge with deps=gcc but no depfile makes no sense");
+      *err = string("edge with deps=") + deps_type +
+             string("but no depfile makes no sense");
```

Comment:
Inefficient string concatenation.

At the very minimum, use .append().

Reply:
Changed to use append().